### PR TITLE
auth: Add some tracing to help debug Frontegg issues

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR adds some additional tracing to our Frontegg auth code which should help us debug future issues.

### Motivation

Help debug issues

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

File: src/frontegg-auth/src/client/tokens.rs [hunk 1]
```diff
@@ -68,7 +92,27 @@ impl Client {
             .with_label_values(&[name, &status])
             .inc();
 
-        let result = response.error_for_status()?.json().await?;
+        let frontegg_trace_id = response
```
Comment (mjibson) [inline src/frontegg-auth/src/client/tokens.rs:95]: This new block is duplicated, can it be a function?

File: src/frontegg-auth/src/auth.rs [hunk 2]
```diff
@@ -476,9 +482,15 @@ fn continuously_refresh(
             // The Frontegg Python SDK scales the expires_in this way.
             //
             // <https://github.com/frontegg/python-sdk/blob/840f8318aced35cea6a41d83270597edfceb4019/frontegg/common/frontegg_authenticator.py#L45>
-            let scaled_expires_in = expires_in * 0.8;
+            let scaled_expires_in = expires_in * 0.5;
```
Comment (mjibson) [inline src/frontegg-auth/src/auth.rs:485]: Justification for decreasing this?

File: src/frontegg-auth/src/error.rs [hunk 3]
```diff
@@ -22,6 +22,8 @@ pub enum Error {
     ReqwestError(Arc<reqwest::Error>),
     #[error("middleware programming error: {0}")]
     MiddlewareError(Arc<anyhow::Error>),
+    #[error("request to Frontegg failed: {0:?}")]
+    FronteggError(Arc<(reqwest::Error, String)>),
```
Comment (mjibson) [inline src/frontegg-auth/src/error.rs:26]: I don't see the second part here, the body, ever being read. Is it needed for something?

File: src/frontegg-auth/src/client/tokens.rs [hunk 4]
```diff
@@ -39,7 +43,27 @@ impl Client {
...
+            .flatten()
+            .map(|trace_id| trace_id.to_string());
+
+        // If we get an error back from Frontegg we want to log the body.
+        if let Err(err) = response.error_for_status_ref() {
+            let body = response
+                .text()
+                .await
+                .unwrap_or("failed to deserialize body".to_string());
+            let error = Arc::new((err, body));
```
Comment (mjibson) [inline src/frontegg-auth/src/client/tokens.rs:59]: Where is the body logged?

**Comment by ParkMyCar** [comment]
Sorry for the noise @mjibson I should have made this a draft PR, thank you for the quick review though, going to iterate

**Comment by ParkMyCar** [comment]
Closing in favor of #25409

**Pull request closed without merging**